src/components/Posts/PostDetails.tsx

```diff
@@ -0,0 +1,62 @@
+import { FC, useState, useEffect } from 'react';
+import { Loader } from '../Loader';
+import { NewCommentForm } from '../Comments/NewCommentForm';
+import { useCommentsContext } from '../../hooks/useCommentsContext';
+import { usePostsContext } from '../../hooks/usePostsContext';
+import { CommentList } from '../Comments/CommentList';
+import { Notification } from '../Notification';
+import { NotificationType } from '../../types';
+import { useGlobalContext } from '../../hooks/useGlobalContext';
+
+export const PostDetails: FC = () => {
+  const { error } = useGlobalContext();
+  const { selectedPost } = usePostsContext();
+  const { isCommentsLoading } = useCommentsContext();
+
+  const [writingNewPost, setWritingNewPost] = useState(false);
```

Comment:
The state variable 'writingNewPost' is not appropriately named for its purpose or usage in the context of post details. It would be better to rename it to something like 'writingNewComment' to match its usage.